src/ui/common/src/reducers/workflow.ts

```diff
@@ -464,8 +448,82 @@ export const handleGetSelectDagPosition = createAsyncThunk<
     const artifactResults =
       thunkAPI.getState().workflowReducer.artifactResults ?? {};
+
     if (!!dag) {
-      return collapsePosition(allNodes, dag, artifactResults);
+      const collapsedPosition = collapsePosition(
+        allNodes,
+        dag,
+        artifactResults
+      );
+
+      const elk = new ELK();
+      const mappedNodes = collapsedPosition.nodes.map((node) => {
+        return {
+          id: node.id,
+          // Width and height set not only the node width/height, but also how far apart we want to position these nodes.
+          // So using a value of just the width and the height of the node will result in no spacing in between.
+          // Elk (eclipse layout kernel) also provides many knobs to tweak this as well.
+          width: 300,
+          height: 300,
+        };
+      });
+
+      // TODO: move into collapsed nodes function.
+      const mappedEdges = collapsedPosition.edges.filter((mappedEdge) => {
+        // Check if the edge exists in the mappedNodes array.
+        // If it does not exist, remove the edge. elk crashes if an edge does not have a corresponding node.
+        const nodeFound = false;
+        for (let i = 0; i < mappedNodes.length; i++) {
+          if (mappedEdge.source === mappedNodes[i].id) {
+            return true;
+          }
+        }
+
+        return nodeFound;
+      });
+
+      const graph = {
+        id: 'root',
+        layoutOptions: {
+          'elk.algorithm': 'layered',
+          'elk.direction': 'RIGHT',
+          'elk.alignment': 'CENTER',
+          'elk.spacing.nodeNode': '80',
+          'elk.layered.spacing.nodeNodeBetweenLayers': '80',
+          // https://www.eclipse.org/elk/reference/options/org-eclipse-elk-layered-nodePlacement-strategy.html
+          'nodePlacement.strategy': 'NETWORK_SIMPLEX',
+          'org.eclipse.elk.edgeRouting': 'SPLINES',
+          //https://www.eclipse.org/elk/reference/options/org-eclipse-elk-layered-nodePlacement-strategy.html
+          'crossingMinimization.strategy': 'INTERACTIVE',
+          'crossingMinimization.forceNodeModelOrder': true,
+        },
+        children: mappedNodes,
+        edges: mappedEdges,
+      };
+
+      console.log('mappedEdges: ', mappedEdges);
+
+      try {
+        const positionedLayout = await elk.layout(graph);
+        collapsedPosition.nodes = collapsedPosition.nodes.map((node) => {
+          for (let i = 0; i < positionedLayout.children.length; i++) {
+            if (node.id === positionedLayout.children[i].id) {
+              node.position = {
+                x: positionedLayout.children[i].x,
+                y: positionedLayout.children[i].y,
+              };
+
+              // TODO (future work): Handle mapping elk edge sections to react-flow edges
+            }
+          }
+
+          return node;
+        });
+      } catch (error) {
+        console.log('error getting PositionedLayout: ', error);
```

Comment:
TODO: change this to show an error alert instead so the user knows that something failed.

Reply:
ENG-2304